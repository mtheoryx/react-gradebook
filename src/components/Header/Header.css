.Header {
  background-color: #222;
  height: 40px;
  padding: 20px;
  color: white;
}
