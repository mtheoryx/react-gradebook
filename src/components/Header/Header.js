import React from 'react';
import './Header.css';
import { titleCase } from '../../helpers/strings';

const Header = props => {
    const text = props.text ? titleCase(props.text) : '';
    return (
        <div className="Header">
            <h2>{text}</h2>
        </div>
    )
};

export default Header;
