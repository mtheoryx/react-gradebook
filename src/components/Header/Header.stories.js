import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Header from './Header';

//Helper just for storybook viewing



storiesOf('Header', module)
    .addDecorator(story => (
        <div style={{textAlign: 'center'}}>
            { story() }
        </div>
    ))
    .add('Default View', () => (<Header />));
