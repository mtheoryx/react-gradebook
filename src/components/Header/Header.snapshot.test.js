import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';

test('Header renders default text value', () => {
    const rendered = renderer.create(
        <Header />
    );

    expect(rendered.toJSON()).toMatchSnapshot();
});

test('Header renders the provided text in title case', () => {
    const rendered = renderer.create(
        <Header text="sample text to caps" />
    );

    expect(rendered.toJSON()).toMatchSnapshot();
});
