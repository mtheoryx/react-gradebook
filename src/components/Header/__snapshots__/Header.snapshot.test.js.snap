exports[`test Header renders default text value 1`] = `
<div
  className="Header">
  <h2>
    Sample Text
  </h2>
</div>
`;

exports[`test Header renders the provided text in title case 1`] = `
<div
  className="Header">
  <h2>
    Sample Text To Caps
  </h2>
</div>
`;
